docs/DeveloperGuide.md

```diff
@@ -1,34 +1,956 @@
+
+### 4.4. Storage (Neil)
+
+The `Storage` class is used to read and write the application state to and from a text file.
+
+#### 4.4.1. Storage format
+
+This section describes the format used to store the TaskList and NotebookShelf from the application state in a plain
+text file.
+
+:exclamation: The operating system newline character is used to terminate lines; that is, `\r\n` on Windows and
+`\n` on UNIX-based systems.
+
+##### 4.4.1.1. TaskList
+
+For `TaskList`, the format is as follows:
+
+* One line containing an integer `n`, the number of tasks.
+* `3n` lines follow, every 3 lines describing one task.
+* The first line contains the task title as a String.
+* The second line contains the task due date, in the format `dd-MM-yyyy hhmm`.
+* The third line contains a value `true` or `false`, indicating whether the task is done. (`true` means done,
+`false` means not done.)
+
+##### 4.4.1.2. Page
+
+For a single page, the format is as follows:
+
+* One line containing the title of the page.
+* One line containing the content of the page. Newline characters in the content are replaced with the string `"~~~"`.
+
+##### 4.4.1.3. Section
+
+For a single section, the format is as follows:
+
+* One line containing an integer `p`, the number of pages in the section.
+* `p` pages are then described, as specified [here](#4412-page).
+
+##### 4.4.1.4. Notebook
+
+For a single notebook, the format is as follows:
+
+* One line containing an integer `s`, the number of sections.
+* `s` sections are then described, as specified [here](#4413-section)
+
+##### 4.4.1.5. NotebookShelf
+
+For `NotebookShelf`, the format is as follows:
+
+* One line containing an integer `n`, the number of notebooks.
+* `n` notebooks are then described, as specified [here](#4414-notebook).
+
+#### 4.4.2 Implementation
+
+`TaskList`, `Task`,`NotebookShelf`, `Notebook`, `Section`, and `page` contain methods called `serialize()`.
+These methods return`String` representations of themselves, as specified [here](#441-storage-format).  
+
+The `serialize()` method in `TaskList` calls the `serialize()` methods for every `Task` object within
+it and combines their outputs with a `StringBuilder`.
+
+The `serialize()` method in `NotebookShelf` calls the `serialize()` methods for every `Notebook` object within it and
+combines their outputs with a `StringBuilder`.
+
+The `serialize()` method in `Notebook` calls the `serialize()` methods for every `Section` object within
+it and combines their outputs with a `StringBuilder`.
+
+The `serialize()` method in `Section` calls the `serialize()` methods for every `Page` object within it and
+combines their outputs with a `StringBuilder`.
+
+##### 4.4.2.1. Saving the application state
+
+The following sequence diagram describes the operation of the `saveToFile()` operation
+
+![Sequence Diagram for saveToFile command](diagrams/class/jpeg/storage_neil.jpg)
```

Comment:
This diagram seems too colorful. You could try out another color instead of pink. Furthermore, the methods should be labeled on top of the arrows. Right now, some of the methods completely cover up the arrow.
![image](https://user-images.githubusercontent.com/46095141/97396648-735d2a00-1922-11eb-852b-709335e5f246.png)

Reply:
Do you think I should leave it to be the same blue as the rest of the blocks?

Noted on the labels, that was what Visio did by default and I didn't think to change it.